### PR TITLE
document and pin SQS version to 2.21.15

### LINKED ISSUE #472: Version 3.1.0 is incompatible with Micronaut 4.2.x
### Expected Behavior

Consistent AWS SDK should be used between Micronaut and its referenced version of micronaut-jms.

### Actual Behaviour

Different versions of the AWS SDK are used, which leads to errors:

```
[...] Caused by: java.lang.NoClassDefFoundError: software/amazon/awssdk/protocols/query/interceptor/QueryParametersToBodyInterceptor
	at software.amazon.awssdk.services.sns.DefaultSnsBaseClientBuilder.finalizeServiceConfiguration(DefaultSnsBaseClientBuilder.java:73)
	at software.amazon.awssdk.awscore.client.builder.AwsDefaultClientBuilder.finalizeChildConfiguration(AwsDefaultClientBuilder.java:185)
	at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.syncClientConfiguration(SdkDefaultClientBuilder.java:193)
	at software.amazon.awssdk.services.sns.DefaultSnsClientBuilder.buildClient(DefaultSnsClientBuilder.java:39)
	at software.amazon.awssdk.services.sns.DefaultSnsClientBuilder.buildClient(DefaultSnsClientBuilder.java:28)
	at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.build(SdkDefaultClientBuilder.java:157)
	at io.micronaut.aws.sdk.v2.service.AwsClientFactory.syncClient(AwsClientFactory.java:110)
	at io.micronaut.aws.sdk.v2.service.sns.SnsClientFactory.syncClient(SnsClientFactory.java:80)
	at io.micronaut.aws.sdk.v2.service.sns.$SnsClientFactory$SyncClient1$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2309)
	... 124 more
```

### Steps To Reproduce

_No response_

### Environment Information

_No response_

### Example Application

_No response_

### Version

4.2.3

### PR DESCRIPTION
**Pull request by sdelamo**
Close #472

### CODE REVIEW

**Comment by sdelamo** [comment]
> Looks like reflection may have changed 🤔
> 
> https://ge.micronaut.io/s/hsfku4qhrj6ka/console-log?page=1#L268
> 
> ```
> Message: org.graalvm.nativeimage.MissingReflectionRegistrationError: The program tried to reflectively invoke method public int org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.getCompressionLevel() without it being registered for runtime reflection. Add it to the reflection metadata to solve this problem. See https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection for help.
> ```

I added the reflection configuration.

File: tests/tasks-sqs/build.gradle.kts [hunk 1]
```diff
@@ -13,3 +13,8 @@ micronaut {
         additionalModules.add("localstack-sqs")
     }
 }
+configurations.all {
+    resolutionStrategy {
+        force("${libs.aws.sqs.get()}")
```
Comment (timyates) [inline tests/tasks-sqs/build.gradle.kts:18]: The tests here pass without this, is that expected?

File: src/main/docs/guide/configuration/sqs.adoc [hunk 2]
```diff
@@ -14,6 +14,16 @@ and the following to your `build.gradle` or `pom.xml`
...
+Force the `software.amazon.awssdk:sqs` dependency to `2.21.15`. For example, with gradle:
+
+[source,kotlin]
+----
+configurations.all {
+    resolutionStrategy {
+        force("software.amazon.awssdk:sqs:2.21.15")
+    }
+}
+----
```
Comment (timyates) [inline src/main/docs/guide/configuration/sqs.adoc:27]: Will we need to update this every time we update aws or sqs?

How will we make sure we don't forget?

Comment (sdelamo) [reply src/main/docs/guide/configuration/sqs.adoc:27]: I want to [do a major release of the module and update to `Jakarta.jms`](https://github.com/micronaut-projects/micronaut-jms/pull/474), and it should work without the need to pin to an old version. We should be able to remove this doc snippet and the forced pin.

File: tests/tasks-sqs/build.gradle.kts (hunk 1 above)
Comment (sdelamo) [reply tests/tasks-sqs/build.gradle.kts]: It fails for me locally https://ge.micronaut.io/s/4lgnag3g2cfca

**Pull request merged**